app/scripts/modules/oracle/subnet/subnet.read.service.ts

```diff
@@ -0,0 +1,65 @@
+import {module} from 'angular';
+import {INFRASTRUCTURE_CACHE_SERVICE, InfrastructureCacheService} from 'core/cache/infrastructureCaches.service';
+import {API_SERVICE, Api} from 'core/api/api.service';
+
+export interface ISubnet {
+  id: string;
+  name: string;
+  account: string;
+  region: string;
+  type: string;
+  vcnId: string;
+  availabilityDomain: string;
+  securityListIds: [string];
+  label: string;
+  purpose: string;
+  deprecated: boolean;
+}
+
+export class SubnetReader {
```

Comment:
I don't see too many differences between this reader and the core subnet reader - could you use that one instead? 

You can create an interface `IOrcaleBMCSSubnet` that extends `ISubnet`, which you can use to type values returned from the core subnet reader.

Also, if you use a different cache namespace, your subnets will not be refreshed when you click the `Refresh` button inside the `Cache Management` section on the application config screen.